### PR TITLE
IvI-2250 Add Uphold Support

### PR DESCRIPTION
**Pull request by erik-mythical**
Not really sure about what this needs to look like so I decided to start the PR even though I'm still working on the implementation in ivi-engine.

### CODE REVIEW

File: api/order/payment.proto [hunk 1]
```diff
@@ -10,4 +10,5 @@ enum PaymentProviderId {
   BRAINTREE = 0;
   BITPAY = 1;
   CYBERSOURCE = 2;
+  UPHOLD = 6;
```
Comment (keith-miller) [inline api/order/payment.proto:13]: Why 6?

**Review by keith-miller — CHANGES_REQUESTED** [review]
Just one question

File: api/order/payment.proto (hunk 1 above)
Comment (erik-mythical) [reply api/order/payment.proto:13]: That is the ordinal in the platform-grpc enum of the same thing and I wasn't sure if we wanted to use the same ordinal here or not.

File: api/wallet/rpc.proto [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+option java_multiple_files = true;
+option java_package = "games.mythical.ivi.sdk.proto.api.wallet";
+
+package ivi.rpc.api.wallet;
+
+import "api/wallet/definition.proto";
+
+service WalletService {
+  rpc GetWalletUser(ivi.proto.api.wallet.GetWalletUserRequest) returns (ivi.proto.api.wallet.Wallet);
```
Comment (kwang-mythical) [inline api/wallet/rpc.proto:11]: are you starting only with the GET and working thru later to figure out what else you need or do you have the necessary calls figured out?

Comment (erik-mythical) [reply api/wallet/rpc.proto:11]: I'm not entirely sure what is needed in the SDK so I'm starting with the GET. I think the changes in this PR are what is needed to view the Uphold wallet data and make payments with an Uphold wallet.

File: api/order/payment.proto (hunk 1 above)
Comment (keith-miller) [reply api/order/payment.proto:13]: Nah, let's make it 3 please.

File: api/wallet/rpc.proto (hunk 2 above)
Comment (keith-miller) [reply api/wallet/rpc.proto:11]: I think we just need to support the game/web stores to allow users to use existing wallets/balances for payment. We don't need to add the functionality to create/update/delete. Users can go to their account page in marketplace to do that. Adding @sebastiano-mythical to confirm

**Review by keith-miller — APPROVED** [review]
lgtm

File: api/wallet/rpc.proto (hunk 2 above)
Comment (sebastiano-mythical) [reply api/wallet/rpc.proto:11]: I'll have an answer on this once the UI designs for the webstore are completed. They have an "account" page and the initial direction is to point them to the marketplace wallet page to link, so Get would be the only required command.
However I'll circle back once execs have approved the store UI flow.

Comment (keith-miller) [reply api/wallet/rpc.proto:11]: Alright, if it changes then that can be another task for later. For now we'll just focus on using wallets for payments.

Comment (erik-mythical) [reply api/wallet/rpc.proto:11]: Sounds good. Thanks for the clarification.

**Review by kwang-mythical — APPROVED** [review]

**Pull request merged**